tools/sof_perf_analyzer.py

```diff
@@ -204,8 +205,29 @@ def process_kmsg_file():
                 next_line = next(f)
                 widget_id = next_line.split('|')[0].split(':')[-1].strip()
                 # convert to the same ID format in mtrace
-                widget_id = int('0x' + widget_id[-6:], 16)
+                try:
+                    widget_id = int('0x' + widget_id[-6:], 16)
+                except ValueError:
+                    print("no valid widget ID found, try with bind information")
+                    not_found[widget_name] = ppln_id
+                    continue
                 comp_name[str(Component(ppln_id, widget_id))] = widget_name
+            elif match_obj := re.search(r' bind ', line):
+                if not_found:
+                    # parse current line to get widget name
+                    span_end_pos = match_obj.span()[1] - 1
+                    line_split = line[span_end_pos + 1:].split()
+                    widget_name = line_split[0].split(':')[0]
```

Comment:
we got two widget_name (s) and two widget_id (s) from the bind message, you need to handle both, what if the not_found widget is the second widget in bind message?

Reply:
the code are making sure we are getting the first widget name and first widget ID.
as you see, there is [0] for widget_name, we always use first to match the case.